ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/show/status/formatter/ShowTableStatusFormatter.java

```diff
@@ -0,0 +1,193 @@
+    if (MetaDataFormatUtils.isJson(conf)) {
+      return new JsonShowTableStatusFormatter();
+    } else {
+      return new TextShowTableStatusFormatter();
+    }
+  }
+
+  public abstract void showTableStatus(DataOutputStream out, Hive db, HiveConf conf, List<Table> tables, Partition par)
+      throws HiveException;
+
+  StorageInfo getStorageInfo(Table table, Partition partition) throws HiveException {
+    String location = null;
+    String inputFormatClass = null;
+    String outputFormatClass = null;
+    if (partition != null) {
+      if (partition.getLocation() != null) {
+        location = partition.getDataLocation().toString();
+      }
+      inputFormatClass = partition.getInputFormatClass() == null ? null : partition.getInputFormatClass().getName();
+      outputFormatClass = partition.getOutputFormatClass() == null ? null : partition.getOutputFormatClass().getName();
+    } else {
+      if (table.getPath() != null) {
+        location = table.getDataLocation().toString();
+      }
+      inputFormatClass = table.getInputFormatClass() == null ? null : table.getInputFormatClass().getName();
+      outputFormatClass = table.getOutputFormatClass() == null ? null : table.getOutputFormatClass().getName();
+    }
+
+    return new StorageInfo(location, inputFormatClass, outputFormatClass);
+  }
+
+  final static class StorageInfo {
+    final String location;
+    final String inputFormatClass;
+    final String outputFormatClass;
+
+    private StorageInfo(String location, String inputFormatClass, String outputFormatClass) {
+      this.location = location;
+      this.inputFormatClass = inputFormatClass;
+      this.outputFormatClass = outputFormatClass;
+    }
+  }
+
+  List<Path> getLocations(Hive db, Partition partition, Table table) throws HiveException {
+    List<Path> locations = new ArrayList<Path>();
+    if (table.isPartitioned()) {
+      if (partition == null) {
+        for (Partition currPartition : db.getPartitions(table)) {
+          if (currPartition.getLocation() != null) {
+            locations.add(new Path(currPartition.getLocation()));
+          }
+        }
+      } else {
+        if (partition.getLocation() != null) {
+          locations.add(new Path(partition.getLocation()));
+        }
+      }
+    } else {
+      if (table.getPath() != null) {
+        locations.add(table.getPath());
+      }
+    }
+    return locations;
+  }
+
+  FileData getFileData(HiveConf conf, List<Path> locations, Path tablePath) throws IOException {
+    FileData fileData = new FileData();
+    FileSystem fileSystem = tablePath.getFileSystem(conf);
+    // in case all files in locations do not exist
+    try {
+      FileStatus tmpStatus = fileSystem.getFileStatus(tablePath);
+      fileData.lastAccessTime = tmpStatus.getAccessTime();
+      fileData.lastUpdateTime = tmpStatus.getModificationTime();
+    } catch (IOException e) {
+      LOG.warn("Cannot access File System. File System status will be unknown: ", e);
```

Comment:
`"Cannot access File System. File System status will be unknown.`

Reply:
Fixed.